test/extensions/filters/http/ext_proc/ext_proc_integration_test.cc

```diff
@@ -1577,6 +1577,69 @@ TEST_P(ExtProcIntegrationTest, GetAndRespondImmediately) {
   EXPECT_EQ("{\"reason\": \"Not authorized\"}", response->body());
 }
 
+// This is almost the same as GetAndRespondImmediately but the filter is
+// configured as an upstream filter.
+TEST_P(ExtProcIntegrationTest, GetAndRespondImmediately_Upstream) {
+  ConfigOptions config_option = {};
+  config_option.downstream_filter = false;
+
+  initializeConfig(config_option);
+  // Add ext_proc as upstream filter.
```

Comment:
Now there are two tests are creating ext_proc filter in upstream: https://github.com/envoyproxy/envoy/blob/2a5978a7d239358c7e8675ded3ae7a4ca8e63379/test/extensions/filters/http/ext_proc/ext_proc_integration_test.cc#L3759.  Probably making this as function and call it at both places?

Reply:
https://github.com/envoyproxy/envoy/pull/41231/commits/548aadbbfa092f8e7207b9aa375065beea97fec0 done. JFYI the similar code exists in observability_integration(?) but not exactly the same so decided to limit the sharing in this file.